### PR TITLE
Update shopping Item

### PR DESCRIPTION
**Pull request by Wandersmann731**

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -144,12 +159,13 @@ const FormContainer = styled.form`
   }
 
   button {
```
Comment (IsabellUrban) [inline pages/index.js:161]: You can't nest a button in a Button Component. This contradict the semantic html.

**Review by IsabellUrban — COMMENTED** [review]
Hi, I found some things in your code. You use a lot of margin in your styled Components. In many layouts, using too much margin can cause unexpected results. Better use gap or padding instead. The styled components are missing the prefix "styled". We have learned this as a naming convention. For example: const StyledDeleteButton = styled.button`. We should not abbreviate event with e. In the course we wrote functions with function keyword and avoided arrow functions.

File: pages/index.js [hunk 2]
```diff
@@ -286,37 +320,81 @@ const ShoppingList = () => {
     setDeletingItemId(null);
   };
 
+  // const toggleFormVisibility = () => {
+  //   setIsFormVisible((prev) => !prev);
+  // };
```
Comment (NikRoe) [inline pages/index.js]: You should remove code you don't need:

```suggestion
```

File: pages/index.js [hunk 3]
```diff
@@ -286,37 +320,81 @@ const ShoppingList = () => {
...
+
+  const handleCancel = (e) => {
+    e.preventDefault();
+    setIsFormVisible(false);
+    e.target.closest("form").reset();
+  };
```
Comment (NikRoe) [inline pages/index.js]: You don't need this function. Your button would also work if you gave it the attribute `type="reset"` and an `onClick` that only sets the `isFormVisible` state to `false`.

File: pages/index.js [hunk 4]
```diff
@@ -286,37 +320,81 @@ const ShoppingList = () => {
...
+            type="text"
+            id="comment"
+            name="comment"
+            placeholder="Optional"
+            defaultValue={editingItem?.comment || ""}
+          />
+          <EditButton type="submit">
+            {editingItem ? "Save Changes" : "Add Item"}
+          </EditButton>
+          <Button onClick={handleCancel}>Cancel</Button>
```
Comment (NikRoe) [inline pages/index.js]:
```suggestion
          <Button type="reset" onClick={() => setIsFormVisible(false)}>Cancel</Button>
```

File: pages/index.js [hunk 5]
```diff
@@ -165,10 +181,11 @@ const FormContainer = styled.form`
 const DeleteButton = styled.button`
   background-color: #007bff;
   color: white;
-  padding: 10px 15px;
+  padding: 10px ;
   border: none;
   border-radius: 4px;
   cursor: pointer;
+  margin 10px;
```
Comment (NikRoe) [inline pages/index.js]:
```suggestion
  margin: 10px;
```

File: pages/index.js [hunk 6]
```diff
@@ -73,10 +87,11 @@ const Select = styled.select`
 const Button = styled.button`
   background-color: #007bff;
   color: white;
-  padding: 10px 15px;
+  padding: 10px;
   border: none;
   border-radius: 4px;
   cursor: pointer;
+  margin 10px;
```
Comment (NikRoe) [inline pages/index.js]:
```suggestion
  margin: 10px;
```

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hi there,

nice start!

I left some suggestions for you to implement. Please also take a look at the comment from Isabell, since it contains some valuable feedback.

Keep it up 🙌

File: pages/index.js
Comment (LorenzCoachK) [inline pages/index.js:1]: Please mark quantity as required and lay an errormessage if the user deletes the value in quantity.

**Pull request merged**